### PR TITLE
[App] Resolve PythonServer on M1

### PR DESCRIPTION
**Pull request by tchaton**
## What does this PR do?

torch.inference seems to generate random images with Stable Diffusion on M1. Use torch.no_grad instead.

Fixes #\<issue_number>

## Before submitting

- [x] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/Lightning-AI/lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [x] Did you make sure to **update the documentation** with your changes? (if necessary)
- [x] Did you write any **new necessary tests**? (not for typos and docs)
- [x] Did you verify new and **existing tests pass** locally with your changes?
- [x] Did you list all the **breaking changes** introduced by this pull request?
- [x] Did you **update the CHANGELOG**? (not for typos, docs, test updates, or minor internal changes/refactors)

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing, make sure you have read the [review guidelines](https://github.com/Lightning-AI/lightning/wiki/Review-guidelines). In short, see the following bullet-list:

- [x] Is this pull request ready for review? (if not, please submit in draft mode)
- [x] Check that all items from **Before submitting** are resolved
- [x] Make sure the title is self-explanatory and the description concisely explains the PR
- [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?

Make sure you had fun coding 🙃

cc @borda

### CODE REVIEW

File: requirements/app/ui.txt [hunk 1]
```diff
@@ -1,2 +1,2 @@
-streamlit>=1.3.1, <=1.11.1
+streamlit
```
Comment (Borda) [inline requirements/app/ui.txt:1]: what are the versions? at least some mon versions...
```suggestion
streamlit>=1.0.0
```

File: src/lightning_app/frontend/streamlit_base.py [hunk 2]
```diff
@@ -20,6 +21,9 @@ def _get_render_fn_from_environment() -> Callable:
 
 def _main():
     """Run the render_fn with the current flow_state."""
+    loop = asyncio.new_event_loop()
```
[the remark this answers is not in the document]
Comment (tchaton) [reply src/lightning_app/frontend/streamlit_base.py]: Removed. This is added while debugging some issues.

**Review by Borda — APPROVED** [review]

File: src/lightning_app/components/auto_scaler.py [hunk 3]
```diff
@@ -269,9 +267,7 @@ async def sys_info(authenticated: bool = Depends(authenticate_private_endpoint))
 
         @fastapi_app.put("/system/update-servers")
         async def update_servers(servers: List[str], authenticated: bool = Depends(authenticate_private_endpoint)):
-            async with lock:
```
Comment (awaelchli) [inline src/lightning_app/components/auto_scaler.py:272]: How are the changes in auto scaler related to the python server on M1?

File: requirements/app/ui.txt [hunk 4]
```diff
@@ -1,2 +1,2 @@
-streamlit>=1.3.1, <=1.11.1
+streamlit>=1.0.0
```
Comment (awaelchli) [inline requirements/app/ui.txt:1]: I would advise against removing the version limit. Please consider bumping the version instead:
```suggestion
streamlit>=1.0.0, <=1.15.2
```
